src/service/action-impl.js

```diff
@@ -501,38 +508,49 @@ export class ActionService {
    * @param {?ActionEventDef} event
    * @param {!ActionTrust} trust
    * @param {?JsonObject=} opt_args
+   * @return {boolean} True if an invocation attempt was made.
    * @private
    */
   action_(source, actionEventType, event, trust, opt_args) {
     const action =
         this.findAction_(source, actionEventType);
     if (!action) {
-      return;
+      return false;
     }
     // Use a pseudo-UUID to uniquely identify this sequence of actions.
     // A sequence is all actions triggered by a single event.
     const sequenceId = Math.random();
     // Invoke actions serially, where each action waits for its predecessor
     // to complete. `currentPromise` is the i'th promise in the chain.
+    /** @type {?Promise} */
     let currentPromise = null;
+    let actionInvoked = false;
     action.actionInfos.forEach(({target, args, method, str}) => {
       const dereferencedArgs = dereferenceArgsVariables(args, event, opt_args);
       const invokeAction = () => {
         const node = this.getActionNode_(target);
         if (!node) {
           this.error_(`Target "${target}" not found for action [${str}].`);
+          actionInvoked = false;
           return;
         }
         const invocation = new ActionInvocation(node, method,
             dereferencedArgs, source, action.node, event, trust,
             actionEventType, node.tagName || target, sequenceId);
+        // Whether an action was invoked. An action could be absent in a
+        // whitelist or not invoked because of a nonexistent method. This
+        // is only used to determine if an invocation attempt was made and not
+        // whether it was successful or not.
+        actionInvoked = true;
```

Comment:
For simplicity, I don't think we need `actionInvoked`. Let's just `return true` at the bottom of the function and define this return value as "does an action exist for this event on the given element" rather than "was that action successfully invoked". Make sense?

Reply:
That's definitely simpler but maybe we should at least check that there are actions to invoke? What do you think about returning either action.actionInfos.length > 0 or !!action.

Reply:
Don't we already have that with `if (!action)` on line 518?

Reply:
Ah right, in that case what do you think about returning action.actionInfos.length > 0 at the end of function? Rather than just 'true'

Reply:
I think in practice `action.actionInfos.length` must be >= 1.